libraries/standard/mqtt/src/iot_mqtt_serialize.c

```diff
@@ -522,12 +522,14 @@ static uint8_t * _encodeUserName( uint8_t * pBuffer,
                 if( ( pConnectInfo->userNameLength + AWS_IOT_METRICS_USERNAME_LENGTH ) <= UINT16_MAX )
                 {
                     /* Write the high byte of the combined length. */
-                    *( pBuffer++ ) = UINT16_HIGH_BYTE( ( pConnectInfo->userNameLength +
-                                                         AWS_IOT_METRICS_USERNAME_LENGTH ) );
+                    *pBuffer = UINT16_HIGH_BYTE( ( pConnectInfo->userNameLength +
+                                                   AWS_IOT_METRICS_USERNAME_LENGTH ) );
+                    pBuffer++;
 
                     /* Write the low byte of the combined length. */
-                    *( pBuffer++ ) = UINT16_LOW_BYTE( ( pConnectInfo->userNameLength +
-                                                        AWS_IOT_METRICS_USERNAME_LENGTH ) );
+                    *pBuffer = UINT16_LOW_BYTE( ( pConnectInfo->userNameLength +
+                                                  AWS_IOT_METRICS_USERNAME_LENGTH ) );
+                    pBuffer++;
```

Comment:
This change does increase the complexity to 8, but this is acceptable as the requirement is <= 8 and not < 8.